src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyDateTypeCheck.java

```diff
@@ -0,0 +1,33 @@
+package com.devonfw.ide.sonarqube.common.impl.check.thirdparty;
+
+import com.devonfw.ide.sonarqube.common.api.JavaType;
+import com.devonfw.ide.sonarqube.common.impl.check.DevonArchitecture3rdPartyCheck;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+
+/**
+ * {@link DevonArchitecture3rdPartyCheck} verifying that proper date types are used.
+ */
+@Rule(key = "E7", name = "devonfw 3rd Party Date Type Check", //
+    priority = Priority.MINOR, tags = { "architecture-violation", "devonfw", "thirdparty" })
+public class DevonArchitecture3rdPartyDateTypeCheck extends DevonArchitecture3rdPartyCheck {
+
+    @Override protected String checkDependency(JavaType source, JavaType target) {
```

Comment:
Annotations (like @Override) should be placed one line above its method / class.

Also the indentation is still at 4 spaces. I am guessing that you set your indentation size to 2 spaces, but didn't set the 'Tab policy' to 'Spaces only'. GitHub only recognizes that there is a tab at the start of the line and automatically parses it to 4 spaces.
Easiest would be to just use the devonfw-ide or import the devonfw formatter config into your preferred IDE.

Reply:
Yes, I used this devonfw formatter in intellij, but something didn't work, I will check this.